### PR TITLE
rules: fix convert when rule is None

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

File: mergify_engine/tests/unit/rules/test_convert.py [hunk 1]
```diff
@@ -206,3 +206,41 @@ def test_convert_rebase_fallback():
...
+                        'required_approving_review_count': 1
+                    },
+                    'required_status_checks': {
+                        'contexts': ['continuous-integration/travis-ci'],
+                    }},
+                'merge_strategy': {
+                    'method': 'rebase',
+                }},
+            'branches': {
+                'gh-pages': None,
```
Comment (sileht) [inline mergify_engine/tests/unit/rules/test_convert.py]: In such case the new version must have `base!=gh-pages`.

**Comment by jd** [comment]
Pull-request updated, HEAD is now e91f3becc34c2d2ae564c331797b470dd5ec389c

**Comment by jd** [comment]
Sorry, missed your comment.

**Comment by jd** [comment]
Pull-request updated, HEAD is now dda846d349807eff221e700283cc83f763c8974a

**Comment by jd** [comment]
Pull-request updated, HEAD is now 1fbbca7ceb48e2eb444a7f6e446df08bc5b15e5a

**Comment by jd** [comment]
Pull-request updated, HEAD is now 974b56b672011acf3e6dfecaee3626472258bece

**Comment by jd** [comment]
Pull-request updated, HEAD is now d16f35e7b4d6e7cae29401024ec388ddcd8206ce

**Pull request merged**